pi5/board.go

```diff
@@ -159,19 +149,109 @@ func (b *pinctrlpi5) Reconfigure(
 		return err
 	}
 
-	b.mu.Lock()
-	defer b.mu.Unlock()
+	// make sure every pin has a name. We already know every pin has a pin
+	for _, c := range newConf.Pins {
+		if c.Name == "" {
+			c.Name = c.Pin
```

Comment:
will this actually update `newConf` to include the names? because I'm assuming that's using in `addUserDefinedNames` on line 159

Reply:
this will update the names in newConf, but these names will not actually be used in `addUserDefinedNames`. `addUserDefinedNames` only looks at names that are not hardware pin numbers/bcom labels